File: Programming_Basics_with_Python/nested_loop_lab/04._sum_of_two_numbers.py
interval_start = int(input())
interval_end = int(input())
megical_number = int(input())
combination_counter = 0
flag = False
for x in range(interval_start, interval_end + 1):
    for y in range(interval_start, interval_end + 1):
        combination = x + y
        combination_counter += 1
        if combination == megical_number:
            print(f"Combination N:{combination_counter} ({x} + {y} = {megical_number})")
            flag = True
            break
    if flag:
        break
if not flag:
    print(f"{combination_counter} combinations - neither equals {megical_number}")

File: Programing_Basics_with_JS/nestedLoopsLab/04.SumOfTwoNumbers.js
function sumOfTwoNumbers(input) {
    let startNumber = Number(input[0]);
    let endNumber = Number(input[1]);
    let magicNumber = Number(input[2]);

    let combinationCount = 0;
    let firstNumber = 0;
    let secondNumber = 0;
    let findCombination = false;

    for(let num1 = startNumber; num1 <= endNumber; num1++) {
        if(findCombination) {
            break;
        }
        for(let num2 = startNumber; num2 <= endNumber; num2++){
            combinationCount++;
            if(num1 + num2 === magicNumber) {
                firstNumber = num1;
                secondNumber = num2;
                findCombination = true;
                break;
            }
        }
    }
    if(findCombination) {
        console.log(`Combination N:${combinationCount} (${firstNumber} + ${secondNumber} = ${magicNumber})`);
    } else {
        console.log(`${combinationCount} combinations - neither equals ${magicNumber}`);
    }
}

sumOfTwoNumbers(["1",
"10",
"5"])

sumOfTwoNumbers(["23",
"24",
"20"])

sumOfTwoNumbers(["88",
"888",
"1000"])

sumOfTwoNumbers(["88", 
"888", 
"2000"])

